.github/workflows/build-test-mac.yml

```diff
@@ -17,6 +17,11 @@ jobs:
         with:
           persist-credentials: false
 
+      - name: Install Python
+      - uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236 # v4.7.1
+        with:
+          python-version: '3.11'
```

Comment:
This is fixing recent break at python/node-gyp integration.
https://github.com/nodejs/node-gyp/issues?q=is%3Aissue+ModuleNotFoundError%3A+No+module+named+%27distutils%27+is%3Aclosed